### PR TITLE
New sample and some corrections

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: samples/x11/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+if (X11_FOUND)
+    add_executable(pong-x11
+        main.c
+    )
+    target_link_libraries(pong-x11
+        X11
+        libpong
+    )
+else ()
+    message(STATUS "x11 sample application not being built: x11 not found")
```
Comment (dakerfp) [inline samples/x11/CMakeLists.txt]: Typo put X11

File: samples/x11/main.c [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+   do {
+
+      cmd[0] = 0;
+      cmd[1] = 0;
+
+      while(XPending(d))
+      {
+            XNextEvent(d, &e);
+            if (e.type == KeyPress)
```
Comment (dakerfp) [inline samples/x11/main.c]: Usu if ( !=  ) continue tro avoid nesting code.

File: samples/x11/main.c [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                }
+            }
+      }
+
+      /* Refresh screen */
+      XClearWindow(d, w);
+
+      // Score
+      char msg[3];
+      itoa(g.players[0].score, msg, 3);
```
Comment (dakerfp) [inline samples/x11/main.c]: 3 -> sizeof(msg)

File: samples/x11/main.c [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+   XMapWindow(d, w);
+
+   int cmd[2];
+
+   do {
+
+      cmd[0] = 0;
+      cmd[1] = 0;
+
+      while(XPending(d))
```
Comment (dakerfp) [inline samples/x11/main.c]: Be consistent with {

**Comment by dakerfp** [comment]
Somehow the ball is visually passing through the player bar.

**Comment by dakerfp** [comment]
Handle or disable resizes.

File: samples/x11/CMakeLists.txt (hunk 1 above)
Comment (patrickelectric) [inline samples/x11/CMakeLists.txt]: ✔

File: samples/x11/main.c (hunk 2 above)
Comment (patrickelectric) [inline samples/x11/main.c]: ✔

File: samples/x11/main.c (hunk 3 above)
Comment (patrickelectric) [inline samples/x11/main.c]: ✔

File: samples/x11/main.c (hunk 4 above)
Comment (patrickelectric) [inline samples/x11/main.c]: ✔

**Comment by patrickelectric** [comment]
What kind of resizes are you talking about ? Can be mode specific ?

**Comment by dakerfp** [comment]
The window resizing events issued by X.

**Comment by patrickelectric** [comment]
@dakerfp: X11 sample update with resize event

**Pull request merged**